src/GooglePlacesApi/Services/GooglePlacesApiService.cs

```diff
@@ -40,15 +49,57 @@ public async Task<Predictions> GetPredictionsAsync(string searchText)
             return await _api.GetAutocompleteAsync(searchText, _settings.CreateQueryStringParameters())
                                  .ConfigureAwait(false);
         }
-
-
-        public async Task<Details> GetDetailsAsync(string placeId)
-        {
+
+        public async Task<Details> GetDetailsAsync(string placeId, string sessionToken, DetailLevel detailLevel = DetailLevel.Basic)
+        {
             if (string.IsNullOrWhiteSpace(placeId))
                 throw new ArgumentNullException(nameof(placeId));
 
-            return await _api.GetDetailsAsync(_settings.ApiKey, placeId)
-                             .ConfigureAwait(false);
-        }
-    }
-}
+            string fields = null;
+
+            switch (detailLevel)
+            {
+                case DetailLevel.Basic:
+                    fields = "id,place_id,name,geometry,formatted_address,reference,icon,address_components,adr_address,types,photos,plus_code,scope,url,utc_offset,vicinity,permanently_closed";

```

Comment:
Maybe refactor this to a slightly different format?

basic = "id,place,blablabla";

case DetailLevel.Contact:
fields = $"{basic},extra1,extra2";
break;

case DetailLevel.Atmosphere:
fields = $"{basic},extra3,extra4}

maybe also store the (magic) strings in the Constants file.

Reply:
in peudo-code, something like this:

basic = Constants.BasicRequestFields;
contact = $"{Constants.BasicRequestFields},{Constants.ContactRequestFields}";

Reply:
Done, stored the field strings in the Constants file, then combining them in the GooglePlacesApiService file.